[WTF] Separate SequesteredAllocator::Arena logic from parent

Achierius opened the pull request:
#### 25afb40a178d8b8f8af4fadc9870d38dea2eb6d5
<pre>
[WTF] Separate SequesteredAllocator::Arena logic from parent
<a href="https://bugs.webkit.org/show_bug.cgi?id=289491">https://bugs.webkit.org/show_bug.cgi?id=289491</a>
<a href="https://rdar.apple.com/146690839">rdar://146690839</a>

Reviewed by Yusuke Suzuki.

Previously, the Arena had to hardcode layout information from
SequesteredArenaAllocator so that it could find its parent allocator.
This is safe because we know that each Arena only ever exists within
a SequesteredArenaAllocator, but the logic properly belongs inside the
SAA itself so this patch moves it there.

Symmetrically, the SequesteredArenaAllocator was handling logic that
rather belongs inside of the Arena itself. Putting this there will
enable the usage of multiple separate Arenas w/in a single allocator
in the future.

* Source/WTF/wtf/SequesteredAllocator.h:

Canonical link: <a href="https://commits.webkit.org/291969@main">https://commits.webkit.org/291969@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af3d4b3be61920afd3be25881d280083a809d461

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94554 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14146 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3945 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99574 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45069 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14444 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22574 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/72149 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29466 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97556 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10755 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/85376 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52481 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10450 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3078 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44393 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/87235 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/80666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/3184 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101617 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/93201 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21610 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/15768 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81154 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21860 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81399 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80528 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20077 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25087 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/2473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/14829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21587 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/26716 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/115868 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21266 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/33197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24729 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23004 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/WTF/wtf/SequesteredAllocator.h:
```diff
@@ -121,6 +128,33 @@ class alignas(128) SequesteredArenaAllocator {
+        GranuleList reset(TopGranulePolicy retain)
+        {
+            dataLogFIf(verbose,
+                "Allocator %d: Arena %d: ending lifetime: granule is (%p), allocHead (%p), allocBound (%p)\n",
```
Let's avoid using `dataLogFIf`, and use `dataLogLnIf`.

Achierius replied:
Oops, missed this when rebasing from the old patch-set. Thanks!

Achierius committed:
[WTF] Separate SequesteredAllocator::Arena logic from parent
https://bugs.webkit.org/show_bug.cgi?id=289491
rdar://146690839

Reviewed by Yusuke Suzuki.

Previously, the Arena had to hardcode layout information from
SequesteredArenaAllocator so that it could find its parent allocator.
This is safe because we know that each Arena only ever exists within
a SequesteredArenaAllocator, but the logic properly belongs inside the
SAA itself so this patch moves it there.

Symmetrically, the SequesteredArenaAllocator was handling logic that
rather belongs inside of the Arena itself. Putting this there will
enable the usage of multiple separate Arenas w/in a single allocator
in the future.

* Source/WTF/wtf/SequesteredAllocator.h:

Canonical link: https://commits.webkit.org/291969@main
